### PR TITLE
Use mkdir -p rather than install -D

### LINKED ISSUE #2: Install fails with "illegal option -- D"
Following the instructions in the README, I cloned and ran `make install`:

```
/Users/caleb/code ❯ git clone pbrisbin/ghu
Cloning into 'ghu'...
remote: Counting objects: 142, done.
remote: Total 142 (delta 0), reused 0 (delta 0)
Receiving objects: 100% (142/142), 48.39 KiB | 0 bytes/s, done.
Resolving deltas: 100% (55/55), done.
Checking connectivity... done.
/Users/caleb/code ❯ cd ghu
ghu(master =) ❯ make install
install -D -m755 bin/ghu /usr/local/bin/ghu
install: illegal option -- D
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...
make: *** [install] Error 64
```

### PR DESCRIPTION
**Pull request by calebhearth**
The install binary on OSX does not support -D, so this is potentially more
portable (and worked on my mac).

Resolves #2

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,11 +4,13 @@ test:
...
-	install -D -m755 share/ghu/parse-header $(DESTDIR)$(PREFIX)/share/ghu/parse-header
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(PREFIX)/share/ghu
+	install -m755 bin/ghu $(DESTDIR)$(PREFIX)/bin/ghu
+	install -m755 share/ghu/parse-header $(DESTDIR)$(PREFIX)/share/ghu/parse-header
 
 uninstall:
 	$(RM) $(DESTDIR)$(PREFIX)/bin/ghu \
-	  $(DESTDIR)$(PREFIX)/share/ghu/parse-header
+		$(DESTDIR)$(PREFIX)/share/ghu/parse-header
```
Comment (pbrisbin) [inline Makefile]: Is this also a portability thing? My understanding is it's not necessary and I don't like.

Comment (calebthompson) [inline Makefile]: Ah, no. My editor just fixes whitespace on write. I'll revert this.

Comment (calebthompson) [inline Makefile]: Wow, so did Git. Had to disable both to undo this change.
